testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketShutdownOutputBySelfTest.java

```diff
@@ -140,5 +219,22 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
         public void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
             queue.offer(msg.readByte());
         }
+
+        private void drainWritabilityQueue() throws InterruptedException {
+            while ((writabilityQueue.poll(100, TimeUnit.MILLISECONDS)) != null) {
+                // Just drain the queue.
+            }
+        }
+
+        void assertWritability(boolean isWritable) throws InterruptedException {
+            try {
+                Boolean writability = writabilityQueue.takeLast();
+                assertEquals(isWritable, writability);
+                // TODO(scott): why do we get multiple writability changes here ... race condition?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)